wallet/src/main/java/bisq/wallet/GrpcWalletService.java

```diff
@@ -0,0 +1,248 @@
+        return toCompletableFuture(futureStub.listUnspent(request))
+                .thenApply(response -> response.getUtxosList().stream()
+                        .map(GrpcUtxo::new)
+                        .collect(Collectors.toList()));
+    }
+
+    @Override
+    public CompletableFuture<String> sendToAddress(Optional<String> passphrase, String address, double amount) {
+        SendToAddressRequest.Builder builder = SendToAddressRequest.newBuilder();
+        passphrase.ifPresent(builder::setPassphrase);
+        builder.setAddress(address);
+        builder.setAmount(amount);
+        SendToAddressRequest request = builder.build();
+
+        return toCompletableFuture(futureStub.sendToAddress(request))
+                .thenApply(SendToAddressResponse::getTxId);
+    }
+
+    @Override
+    public CompletableFuture<Boolean> isWalletEncrypted() {
+        IsWalletEncryptedRequest request = IsWalletEncryptedRequest.newBuilder().build();
+        return toCompletableFuture(futureStub.isWalletEncrypted(request))
+                .thenApply(IsWalletEncryptedResponse::getEncrypted);
+    }
+
+    @Override
+    public CompletableFuture<Coin> requestBalance() {
+        GetBalanceRequest request = GetBalanceRequest.newBuilder().build();
+        return toCompletableFuture(futureStub.getBalance(request))
+                .thenApply(response -> {
+                    Coin newBalance = Coin.asBtcFromValue(response.getBalance());
+                    balance.set(newBalance);
+                    return newBalance;
+                });
+    }
+
+    @Override
+    public Observable<Coin> getBalance() {
+        return balance;
+    }
+
+    @Override
+    public ObservableSet<Transaction> getTransactions() {
+        return transactions;
+    }
+
+    @Override
+    public CompletableFuture<ObservableSet<Transaction>> requestTransactions() {
+        ListTransactionsRequest request = ListTransactionsRequest.newBuilder().build();
+        return toCompletableFuture(futureStub.listTransactions(request))
+                .thenApply(response -> {
+                    transactions.clear();
+                    transactions.addAll(response.getTransactionsList().stream()
+                            .map(this::toTransaction)
+                            .toList());
+                    return transactions;
+                });
+    }
+
+    @Override
+    public CompletableFuture<Boolean> shutdown() {
+        channel.shutdown();
+        return CompletableFuture.completedFuture(true);
+    }
+
+    private Transaction toTransaction(bisq.wallets.grpc.pb.Transaction tx) {
+        return new Transaction(
+                tx.getTxId(),
+                tx.getInputsList().stream().map(this::toTransactionInput).toList(),
+                tx.getOutputsList().stream().map(this::toTransactionOutput).toList(),
+                tx.getLockTime(),
+                tx.getHeight(),
+                Optional.of(new Date(tx.getDate() * 1000)),
```

Comment:
I guess its optional as it was used for non confirmed txs and use the block date once mined. I guess its better to remove that and leave it to the client to get the block data. But we can leave it for later when implementation requirements are more clear.